Added support for picklists.

arturocuya opened the pull request.

arturocuya committed:
Added support for picklists.

arturocuya committed:
Added a comment

vchernetsky commented on CacheManager.py:
```diff
@@ -0,0 +1,290 @@
+        return conn
+
+    def closeConn(self,cursor):
+        cursor.close()
```
This closes cursor, but not the connection. It seems that connection is never get closed.

vchernetsky commented on CacheManager.py:
```diff
@@ -0,0 +1,290 @@
+            print(content)
+
+    def openConn(self):
+        conn = sqlite3.connect(self.tempPath)
```
Perhaps conn should be a member of a CacheManager  class. It would be easier to handle the connection, and close it when appropriate. Because not it doesn't get closed at all.

vchernetsky commented on CacheManager.py:
```diff
@@ -0,0 +1,290 @@
+
+    def initializeDB(self):
+        try:
+            conn = self.openConn()
```
This conn doesn't get closed.

vchernetsky commented on CacheManager.py:
```diff
@@ -0,0 +1,290 @@
+        p_img_url += '?token=' + os.environ['AMIGOCLOUD_API_KEY']
+        img_data = urllib.request.urlopen(p_img_url).read()
+        newValues = (p_url,p_id,p_hash,img_data)
+        c.execute("INSERT OR IGNORE INTO projects VALUES (?,?,?,?)", newValues)
```
Shouldn't it be INSERT OR REPLACE?

vchernetsky commented on CacheManager.py:
```diff
@@ -0,0 +1,290 @@
+        ds_img_url += '?token=' + os.environ['AMIGOCLOUD_API_KEY']
+        img_data = urllib.request.urlopen(ds_img_url).read()
+        newValues = (ds_url, ds_p_url, ds_id, ds_hash, img_data)
+        c.execute("INSERT OR IGNORE INTO datasets VALUES (?,?,?,?,?)", newValues)
```
INSERT OR REPLACE

vchernetsky commented on CacheManager.py:
```diff
@@ -0,0 +1,290 @@
+        self.tempPath = os.path.join(self.tempDir, self.dbName)
+        self.isDev = False
+
+    def devPrint(self,content):
```
Names of the functions should not have capital letter, it should look like this: dev_print()

Please see Python naming conventions:
https://www.python.org/dev/peps/pep-0008/#naming-conventions

vchernetsky commented on PicklistManager.py:
```diff
@@ -0,0 +1,22 @@
+                    self.qgm.addValueMap(ds_name,field_name,choices)
+        if raw_choices == None:
+            print('No choices found or picklist marked as not visible')
+            return
```
No new line at the end of file.

vchernetsky commented on amigocloud_dialog.py:
```diff
@@ -23,16 +23,20 @@
     os.path.dirname(__file__), 'amigocloud_dialog_base.ui'))
 
+cm = CacheManager()
+cm.initializeDB()
```
Why is cm is in this global scope, instead of being a member of a class?

arturocuya committed:
Formatted functions and variables' names to fit into PEP 8 standards

vchernetsky commented on QGISManager.py:
```diff
@@ -0,0 +1,58 @@
+from qgis._core import QgsVectorLayer
+from .amigo_api import AmigoAPI
+
+api = AmigoAPI()
```
This should be member of a class.

vchernetsky commented on amigo_api.py:
```diff
@@ -28,13 +29,29 @@ def fetch_project_list(self):
 
     def fetch_dataset_list(self, project_id):
-        dataset_url = self.url + '/api/v1/users/0/projects/' + project_id + '/datasets?summary'
+        dataset_url = self.url + '/api/v1/users/0/projects/' + project_id + '/datasets'
```
Why are we switching away from ?summary? There are implications of this, it makes calls slower, also the page size becomes 20 instead of 100..

vchernetsky commented on amigocloud_dialog.py:
```diff
@@ -50,22 +54,37 @@ def __init__(self, parent=None):
         self.ds_list_widget.itemClicked.connect(self.dataset_clicked)
 
+        self.syncButton = QPushButton('Sync.',self)
+        self.syncButton.move(698,32)
```
Why don't we use UI tool to position the button?

vchernetsky commented on amigocloud_q.py:
```diff
@@ -162,6 +171,7 @@ def initGui(self):
         """Create the menu entries and toolbar icons inside the QGIS GUI."""
 
         icon_path = ':/plugins/amigocloud/icon.png'
+        # icon_path = './rcat.jpg'
```
This line should be deleted

arturocuya committed:
Added file that was missing on the last commit. Changed INSERT OR IGNORE to INSERT OR REPLACE.

arturocuya committed:
Changed even more names of variables and methods to fit PEP8 standards.

arturocuya commented on amigo_api.py:
```diff
@@ -66,7 +81,4 @@ def send_analytics_event(self, category, action, label):
             ejson = json.dumps(e)
             e64 = base64.b64encode(bytes(ejson, 'utf-8'))
             url = 'http://api.mixpanel.com/track/?data=' + str(e64)
-            try:
```
I never touched anything from send_analytics_event. no idea why these lines have been removed but if they were important I will add them again.

vchernetsky replied:
You are correct, those lines should be there.

arturocuya committed:
** LATEST STABLE PUSH **
Modified amigocloud.py urllib imports so they work on py3.
TODO added to amigocloud_dialog.py referring to fix bug caused by the addition of ?summary on a query.
Returned to "initGui" instead of "init_gui" in amigocloud_q.py since the camelCase is necessary for the plugin to run.
Modified the way connection and closing of local database works.

arturocuya committed:
Fixed several bugs in amigocloud.py.  Changed some variable names.

arturocuya committed:
Most of the functionality of the cache relies now on the id as the primary index. Fixed some problem with the local db trash manager not working. Also, now the connection opens every time the user opens the plugin window and closes every time the user presses "Ok" or "Cancel".

arturocuya committed:
Commented obsolete or broken tests.

arturocuya committed:
Modigied gitignore to ignore pytest cache

arturocuya committed:
Moved important files to a 'utils' folder

arturocuya committed:
Added missing init file into utils folder

arturocuya committed:
Created test for picklist manager class.

arturocuya committed:
Remade the CacheManager module.

arturocuya committed:
Improved functions related to update the cache. Added TODOs.

arturocuya committed:
Deleted a line. Moved a method.

vchernetsky commented on utils/CacheManager.py:
```diff
@@ -40,16 +42,16 @@ def format_json_after_fetch(self, string):
+            c = self.database.cursor()
 
-        c.execute("CREATE TABLE IF NOT EXISTS local_cache (url TEXT NOT NULL PRIMARY KEY, name TEXT, schema_hash TEXT, schema TEXT, img_hash TEXT, img BLOB)")
+            c.execute("CREATE TABLE IF NOT EXISTS local_cache (url TEXT NOT NULL PRIMARY KEY, name TEXT, schema_hash TEXT, schema TEXT, img_hash TEXT, img BLOB)")
```
The cache schema should look like this
CREATE TABLE IF NOT EXISTS local_cache (key TEXT NOT NULL PRIMARY KEY, value BLOB)
There should be no schema, or image references.

vchernetsky committed:
Modify cache

vchernetsky committed:
Swap related datasets

vchernetsky committed:
settings_dialog.py

vchernetsky committed:
Add Setting Dialog

vchernetsky committed:
Merge with master

vchernetsky committed:
Unswap the related datasets

vchernetsky committed:
Add Server URL
